timesketch/lib/analyzers/sequence_sessionizer.py

```diff
@@ -0,0 +1,158 @@
+        if self.event_seq is None or self.event_seq == []:
+            raise RuntimeError('No event_seq provided.')
+        # If return_fields in none, then all attributes are provided.
+        if self.return_fields is not None:
+            if 'timestamp' not in self.return_fields:
+                raise RuntimeError('No "timestamp" in return_fields.')
+            if len(self.return_fields) <= 1:
+                raise RuntimeError('No return_fields provided.')
+
+        # event_stream returns an ordered generator of events (by time)
+        # therefore no further sorting is needed.
+        events = self.event_stream(query_string=self.query,
+                                   return_fields=self.return_fields)
+
+        last_timestamp = 0
+
+        try:
+            first_event = next(events)
+            last_timestamp = first_event.source.get('timestamp')
+            self.process_event(first_event)
+
+            for event in events:
+                curr_timestamp = event.source.get('timestamp')
+                if curr_timestamp - last_timestamp > self.max_time_diff_micros:
+                    self.flush_events(drop=True)
+
+                self.process_event(event)
+                last_timestamp = curr_timestamp
+        except StopIteration:
+            pass
+
+        self.query = self.get_query_string()
+
+        self.sketch.add_view('Session view',
+                             self.NAME,
+                             query_string=self.query)
+
+        return ('Sessionizing completed, number of {0:s} session created:'
+                ' {1:d}'.format(self.session_type, self.session_num))
+
+    def annotateEvent(self, event, session_num):
+        """Add an session_type attribute with a session_num to event.
+
+        Args:
+            event: Event to annotate.
+            session_num: Session number for the event.
+        """
+        event.add_attributes({self.session_type: session_num})
+        event.commit()
+
+    def process_event(self, event):
+        """Process event depending on if the event is significant for the
+        searched event sequence.
+
+        Event is significant if it matches the current
+        event in the event_seq or if in the timeline it is between two
+        consistent events from the event_seq.
+
+        Args:
+            event: Event to process.
+        """
+        if self.match_event(event):
+            self.recording = True
+            self.event_storage.append(event)
+            self.num_event_to_find += 1
+
+            if self.num_event_to_find == len(self.event_seq):
+                self.flush_events()
+            return
+
+        if self.recording:
+            self.event_storage.append(event)
+
+    def flush_events(self, drop=False):
+        """Annotates or clears events in event_storage according to the flag
+        drop and resets session recording state.
+
+        Args:
+            drop: Indicator if the stored events should not be commited.
```

Comment:
This sentence is a bit unclear.

```suggestion
            drop: If True, the stored events will not be committed to the database.
```